File: src/main/java/org/example/expert/domain/manager/entity/Log.java
package org.example.expert.domain.manager.entity;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.Setter;

@Entity
@Setter
@Table(name = "log")
public class Log {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private String action;

	private String details;

	private LocalDateTime createdAt;

	@PrePersist
	public void prePersist() {
		this.createdAt = LocalDateTime.now();
	}
}

File: src/main/kotlin/com/example/domain/manager/entity/Log.kt
package com.example.domain.manager.entity

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "log")
class Log(

    val action: String,

    val details: String
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    lateinit var createdAt: LocalDateTime

    @PrePersist // 초기화?
    fun prePersist() {
        this.createdAt = LocalDateTime.now()
    }
}
